kubernetes/annotations_and_labels.md

```diff
@@ -41,18 +41,40 @@ This page defines common annotations and labels we set in Kubernetes objects.
   managed by `chart-operator`.
 - `release.giantswarm.io/version` - value should be Giant Swarm release
   version, e.g. `release.giantswarm.io/version=2.3.0`.
-- `OPERATOR.giantswarm.io/version` - value should be an operator version, e.g.
-  `kvm-operator.giantswarm.io/version=1.0.0`. This is not the same as release
-  version. This value can be the same in multiple releases. This label is used
-  by the operator to recognize which object it should reconcile. When set on
-  nodes it is used to set that information in the status with the
-  statusresource.
+- `OPERATOR.giantswarm.io/version` - value should be the version of the
+  operator as defined in `project.go` and matching `appVersion` in
+  `Chart.yaml`, e.g. `kvm-operator.giantswarm.io/version=1.0.0`. It is used by
+  the given operator to recognize which object it should reconcile (i.e. to
+  only reconcile objects matching its own version). When set on nodes it is
+  used to set that information in the status with the statusresource. This is
+  different from release version and can be the same in multiple releases. Its
+  value may be equal to that of `app.kubernetes.io/version` but it has a
+  different purpose and since there could be multiple operators reconciling one
+  object there could be multiple per-operator labels on one object.
+- `app.kubernetes.io/name`, `app.kubernetes.io/version` and other
+  [common labels][k8s-common-labels] - are informational, usually applied to
+  all objects in a chart that installs an app or operator and so they indicate
+  the source of that app/operator; value of `.../version` should be the
+  app/operator version as defined in `project.go` and matching `appVersion` in
+  `Chart.yaml`, e.g. `app.kubernetes.io/version=1.0.0`.
```

Comment:
Should we mention that here? We have `app` and `version` already. Do you think we should change that in scope of this task? Should we have both? Should we actively migrate to the official recommendation?

Reply:
I wanted to add them for a couple of reasons, mainly that `version` is not the same as `app.kubernetes.io/version` and the second thing was that we decided to drop `app.giantswarm.io/version` which I totally agree as it would just be duplicating the upstream labels, so I feel it's better to mention them here to avoid that. Also there was interest from people to adopt them anyway, but with understanding that they won't replace our internal labels at least not immediately.

Reply:
Yes I think when we introduce something new we should also have a migration plan and actively do this. If you want to do this that's awesome but I'd say it's unrelated to this task.

Reply:
Let's have a chat on google meet on Slack.